### ISSUE TITLE
Crash on iOs 8.3 iPhone 5s during first launch

### ISSUE THREAD

**Issue by chapayGhub** [BranchMetrics/ios-branch-deep-linking#585]
Hi

Guys I need help (((

We have a crash in app this is steps for reproducing:
I can reproduce it only when released app installed as ipa
1 Delete app from device
2 open referal url that will redirect to appstore app's page
3 Install application 
4 launch it
5 Crash

Our App uses Xamarin Branch, but i see in crash log calling form Objc branch code before crash

```
5   Apppp                        	0x1004b15e0 -[BranchOpenRequest processResponse:error:] (BranchOpenRequest.m:132)
6   Apppp                        	0x10049e708 __30-[Branch processNextQueueItem]_block_invoke (Branch.m:1302)
```

This issue already posted to xamarin repository
https://github.com/BranchMetrics/xamarin-branch-deep-linking/issues/48

Thanks!

```
Last Exception Backtrace:
0   CoreFoundation                	0x1839a42d8 __exceptionPreprocess + 132
1   libobjc.A.dylib               	0x1951780e4 objc_exception_throw + 60
2   CoreFoundation                	0x18388d428 -[__NSDictionaryM setObject:forKey:] + 972
3   Apppp                        	0x1004bfae0 +[BNCFabricAnswers prepareBranchDataForAnswers:] (BNCFabricAnswers.m:46)
4   Apppp                        	0x1004bf5a8 +[BNCFabricAnswers sendEventWithName:andAttributes:] (BNCFabricAnswers.m:16)
5   Apppp                        	0x1004b15e0 -[BranchOpenRequest processResponse:error:] (BranchOpenRequest.m:132)
6   Apppp                        	0x10049e708 __30-[Branch processNextQueueItem]_block_invoke (Branch.m:1302)
7   libdispatch.dylib             	0x1957c9994 _dispatch_call_block_and_release + 24
8   libdispatch.dylib             	0x1957c9954 _dispatch_client_callout + 16
9   libdispatch.dylib             	0x1957ce20c _dispatch_main_queue_callback_4CF + 1608
10  CoreFoundation                	0x18395b7f8 __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 12
11  CoreFoundation                	0x1839598a0 __CFRunLoopRun + 1492
12  CoreFoundation                	0x1838852d4 CFRunLoopRunSpecific + 396
13  GraphicsServices              	0x18d0a36fc GSEventRunModal + 168
14  UIKit                         	0x18844afac UIApplicationMain + 1488
15  Apppp                        	0x100842094 wrapper_managed_to_native_UIKit_UIApplication_UIApplicationMain_int_string___intptr_intptr (/<unknown>:1)
16  Apppp                        	0x1007cb714 Xamarin_iOS_UIKit_UIApplication_Main_string___string_string (UIApplication.cs:79)
17  Apppp                        	0x100621cb0 Apppp_Gem_iOS_Application_Main_string__ + 28
18  Apppp                        	0x100af0394 wrapper_runtime_invoke_object_runtime_invoke_dynamic_intptr_intptr_intptr_intptr + 244
19  Apppp                        	0x100506cc0 mono_jit_runtime_invoke (mini-runtime.c:2551)
20  Apppp                        	0x10057f0a0 do_runtime_invoke (object.c:2820)
21  Apppp                        	0x100582438 do_exec_main_checked (object.c:2978)
22  Apppp                        	0x1004eb5ac mono_jit_exec (driver.g.c:1049)
23  Apppp                        	0x100621654 xamarin_main (monotouch-main.m:485)
24  Apppp                        	0x101fedd08 main (main.arm64.m:145)
25  libdyld.dylib                 	0x1957f6a08 start + 4


Thread 0 name:  tid_907  Dispatch queue: com.apple.main-thread
Thread 0:
0   libsystem_kernel.dylib        	0x000000019590f594 __semwait_signal_nocancel + 8
1   libsystem_c.dylib             	0x000000019583f1a0 nanosleep$NOCANCEL + 200
2   libsystem_c.dylib             	0x00000001958642b8 usleep$NOCANCEL + 64
3   libsystem_c.dylib             	0x0000000195886b1c abort + 116
4   Apppp                        	0x000000010047b104 uncaught_exception_handler + 28
5   CoreFoundation                	0x00000001839a4694 __handleUncaughtException + 688
6   libobjc.A.dylib               	0x00000001951783b0 _objc_terminate() + 112
7   Apppp                        	0x00000001004631c0 BITCrashUncaughtCXXTerminateHandler() (BITCrashCXXExceptionHandler.mm:178)
8   libc++abi.dylib               	0x0000000194975bb0 std::__terminate(void (*)()) + 12
9   libc++abi.dylib               	0x0000000194975474 __cxa_throw + 132
10  libobjc.A.dylib               	0x0000000195178200 objc_exception_throw + 344
11  CoreFoundation                	0x000000018388d424 -[__NSDictionaryM setObject:forKey:] + 968
12  Apppp                        	0x00000001004bfadc +[BNCFabricAnswers prepareBranchDataForAnswers:] (BNCFabricAnswers.m:46)
13  Apppp                        	0x00000001004bf5a4 +[BNCFabricAnswers sendEventWithName:andAttributes:] (BNCFabricAnswers.m:16)
14  Apppp                        	0x00000001004b15dc -[BranchOpenRequest processResponse:error:] (BranchOpenRequest.m:132)
15  Apppp                        	0x000000010049e704 __30-[Branch processNextQueueItem]_block_invoke (Branch.m:1302)
16  libdispatch.dylib             	0x00000001957c9990 _dispatch_call_block_and_release + 20
17  libdispatch.dylib             	0x00000001957c9950 _dispatch_client_callout + 12
18  libdispatch.dylib             	0x00000001957ce208 _dispatch_main_queue_callback_4CF + 1604
19  CoreFoundation                	0x000000018395b7f4 __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 8
20  CoreFoundation                	0x000000018395989c __CFRunLoopRun + 1488
21  CoreFoundation                	0x00000001838852d0 CFRunLoopRunSpecific + 392
22  GraphicsServices              	0x000000018d0a36f8 GSEventRunModal + 164
23  UIKit                         	0x000000018844afa8 UIApplicationMain + 1484
24  Apppp                        	0x0000000100842090 wrapper_managed_to_native_UIKit_UIApplication_UIApplicationMain_int_string___intptr_intptr (/<unknown>:1)
25  Apppp                        	0x00000001007cb710 Xamarin_iOS_UIKit_UIApplication_Main_string___string_string (UIApplication.cs:79)
26  Apppp                        	0x0000000100621cac Apppp_Gem_iOS_Application_Main_string__ + 24
27  Apppp                        	0x0000000100af0390 wrapper_runtime_invoke_object_runtime_invoke_dynamic_intptr_intptr_intptr_intptr + 240
28  Apppp                        	0x0000000100506cbc mono_jit_runtime_invoke (mini-runtime.c:2551)
29  Apppp                        	0x000000010057f09c do_runtime_invoke (object.c:2820)
30  Apppp                        	0x0000000100582434 do_exec_main_checked (object.c:2978)
31  Apppp                        	0x00000001004eb5a8 mono_jit_exec (driver.g.c:1049)
32  Apppp                        	0x0000000100621650 xamarin_main (monotouch-main.m:485)
33  Apppp                        	0x0000000101fedd04 main (main.arm64.m:145)
34  libdyld.dylib                 	0x00000001957f6a04 start + 0

Thread 1 name:  Dispatch queue: com.apple.libdispatch-manager
Thread 1:
0   libsystem_kernel.dylib        	0x00000001958f4c24 kevent64 + 8
1   libdispatch.dylib             	0x00000001957d9e6c _dispatch_mgr_invoke + 272
2   libdispatch.dylib             	0x00000001957cb998 _dispatch_mgr_thread + 48

Thread 2 Crashed:
0   libsystem_kernel.dylib        	0x000000019590fc78 __workq_kernreturn + 8
1   libsystem_pthread.dylib       	0x00000001959a92d8 _pthread_wqthread + 988
2   libsystem_pthread.dylib       	0x00000001959a8eec start_wqthread + 0

Thread 3:
0   libsystem_kernel.dylib        	0x000000019590fc78 __workq_kernreturn + 8
1   libsystem_pthread.dylib       	0x00000001959a92d8 _pthread_wqthread + 988
2   libsystem_pthread.dylib       	0x00000001959a8eec start_wqthread + 0

Thread 4 name:  com.apple.NSURLConnectionLoader
Thread 4:
0   libsystem_kernel.dylib        	0x00000001958f4e0c mach_msg_trap + 8
1   libsystem_kernel.dylib        	0x00000001958f4c84 mach_msg + 68
2   CoreFoundation                	0x000000018395b720 __CFRunLoopServiceMachPort + 196
3   CoreFoundation                	0x0000000183959674 __CFRunLoopRun + 936
4   CoreFoundation                	0x00000001838852d0 CFRunLoopRunSpecific + 392
5   CFNetwork                     	0x0000000183366890 +[NSURLConnection(Loader) _resourceLoadLoop:] + 436
6   Foundation                    	0x00000001848a5db4 __NSThread__main__ + 1068
7   libsystem_pthread.dylib       	0x00000001959abdc4 _pthread_body + 160
8   libsystem_pthread.dylib       	0x00000001959abd20 _pthread_start + 156
9   libsystem_pthread.dylib       	0x00000001959a8ef4 thread_start + 0

Thread 5 name:  Dispatch queue: NSOperationQueue 0x17403e1a0 :: NSOperation 0x1744404e0 (QOS: USER_INITIATED)
Thread 5:
0   libsystem_kernel.dylib        	0x000000019590f474 __rmdir + 8
1   libsystem_kernel.dylib        	0x00000001958f865c rmdir + 12
2   Foundation                    	0x00000001847d66fc _NSCleanupTemporaryDirectory + 40
3   Foundation                    	0x00000001847d4e20 _NSWriteDataToFileWithExtendedAttributes + 1472
4   Foundation                    	0x0000000184822da0 -[NSData(NSData) writeToURL:options:error:] + 280
5   Apppp                        	0x00000001004acac4 __41-[BNCPreferenceHelper persistPrefsToDisk]_block_invoke (BNCPreferenceHelper.m:677)
6   Foundation                    	0x000000018488b1c0 __NSBLOCKOPERATION_IS_CALLING_OUT_TO_A_BLOCK__ + 12
7   Foundation                    	0x00000001847dc600 -[NSBlockOperation main] + 92
8   Foundation                    	0x00000001847cc1c8 -[__NSOperationInternal _start:] + 632
9   Foundation                    	0x000000018488df24 __NSOQSchedule_f + 224
10  libdispatch.dylib             	0x00000001957c9950 _dispatch_client_callout + 12
11  libdispatch.dylib             	0x00000001957d40a0 _dispatch_queue_drain + 1444
12  libdispatch.dylib             	0x00000001957cca58 _dispatch_queue_invoke + 128
13  libdispatch.dylib             	0x00000001957d6314 _dispatch_root_queue_drain + 716
14  libdispatch.dylib             	0x00000001957d7c48 _dispatch_worker_thread3 + 104
15  libsystem_pthread.dylib       	0x00000001959a9228 _pthread_wqthread + 812
16  libsystem_pthread.dylib       	0x00000001959a8eec start_wqthread + 0
```

**Comment by E-B-Smith** [comment]
Thanks for all the traces.  I'll look at this and see if I can reproduce this on our end.  When did this problem start?

**Comment by E-B-Smith** [comment]
I added a fix for this in PR #587.
